### PR TITLE
[MI-2032]: Used separate goroutines to delete users when encryption secret is being changed

### PR DESCRIPTION
**Pull request by Nityanand13**

### CODE REVIEW

File: server/store/store.go [hunk 1]
```diff
@@ -68,33 +70,47 @@ func (s *Store) Delete(key string) error {
...
+	wg := new(sync.WaitGroup)
+
 	for {
 		kvList, err := s.api.KVList(page, constants.UsersPerPage)
 		if err != nil {
-			return err
-		}
-		if len(kvList) == 0 {
-			return nil
+			s.api.LogError("Failed to get the users.", "Error", err.Error())
```
Comment (ayusht2810) [inline server/store/store.go:80]:
```suggestion
			s.api.LogError("Failed to get the KVList", "Error", err.Error())
```

**Review by ayusht2810 — APPROVED** [review]

File: server/store/store.go [hunk 2]
```diff
@@ -68,33 +70,48 @@ func (s *Store) Delete(key string) error {
...
-				if err := s.Delete(key); err != nil {
-					return err
+			wg.Add(1)
+
+			go func(key string) {
+				defer wg.Done()
+
+				if userID, isValidUserKey := IsValidUserKey(key); isValidUserKey {
+					isUserDeleted = true
+					if isDeleted, err := s.DeleteUser(userID); !isDeleted {
```
Comment (manojmalik20) [inline server/store/store.go:96]: What's the need for `isUserDeleted` return value? We can just use `error` as the return type.

Comment (Nityanand13) [reply server/store/store.go:96]: Sure

**Comment by Nityanand13** [comment]
PR is made on the mattermost [#145](https://github.com/mattermost/mattermost-plugin-azure-devops/pull/145)

**Pull request closed without merging**